src/gui/network.go

```diff
@@ -26,19 +39,52 @@ func connectionHandler(gateway Gatewayer) http.HandlerFunc {
 			wh.Error404(w, "")
 			return
 		}
+		cnx := Connection{c, 0}
+		bcp := gateway.GetBlockchainProgress()
+		notFound := true
+		i, l := 0, len(bcp.Peers)
+		for notFound && i < l {
+			ph := bcp.Peers[i]
+			if ph.Address == c.Addr {
+				notFound = false
+				cnx.Height = ph.Height
+			}
+			i++
+		}
 
-		wh.SendJSONOr500(logger, w, c)
+		wh.SendJSONOr500(logger, w, cnx)
 	}
 }
 
+func peerHeightIndex(bcp *daemon.BlockchainProgress) (index map[string]uint64) {
+	peerHeights := bcp.Peers
+	index = make(map[string]uint64, len(peerHeights))
+
+	for i := 0; i < len(peerHeights); i++ {
+		index[peerHeights[i].Address] = peerHeights[i].Height
+	}
+	return
+}
+
 func connectionsHandler(gateway Gatewayer) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		if r.Method != http.MethodGet {
 			wh.Error405(w)
 			return
 		}
 
-		wh.SendJSONOr500(logger, w, gateway.GetConnections())
+		dcnxs := gateway.GetConnections()
+		bcp := gateway.GetBlockchainProgress()
+		index := peerHeightIndex(bcp)
+		cnxs := Connections{}
+		for _, c := range dcnxs.Connections {
+			cnx := Connection{c, 0}
+			if height, hasRecord := index[c.Addr]; hasRecord {
```

Comment:
height would be 0 if not found in the map, so it is not necessary to check `hasRecord`